### PR TITLE
feature: support yml configuration

### LINKED ISSUE #2314: feature: registry.conf suffix support customize
## Why you need it?
 Is your feature request related to a problem? Please describe in details
English: The conf component supports reading files of type .conf,.properties,*.yml, but currently only limits reading files of type *.conf when programs depend on seata-all.

Chinese: 用户可以自定义registry.conf的后缀名

## How it could be?
A clear and concise description of what you want to happen. You can explain more about input of the feature, and output of it.
Optimize ConfigurationFactory.java

### PR DESCRIPTION
**Pull request by wangwei-ying**
### Ⅱ. Does this pull request fix one issue?

fix #2314 

### Ⅳ. Describe how to verify it
I used to verify seata-simples 

### Ⅴ. Special notes for reviews
auto load *.conf , *.properties or *.yml  file , according to 'convention over configuration'

### CODE REVIEW

**Comment by l81893521** [comment]
Please solve the code conflict.

**Comment by wangwei-ying** [comment]
> Please solve the code conflict.

done

**Comment by a364176773** [comment]
please solve the code conflict.

**Comment by wangwei-ying** [comment]
> please solve the code conflict.

done

File: config/seata-config-core/src/main/java/io/seata/config/FileConfigFactory.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        {
+            put("conf", DEFAULT_TYPE);
+            put("properties", DEFAULT_TYPE);
+            put("yml", YAML_TYPE);
+        }
+    };
+
+
+    public static FileConfig load() {
+
```
Comment (a364176773) [inline config/seata-config-core/src/main/java/io/seata/config/FileConfigFactory.java:49]: unnecessary

Comment (a364176773) [inline config/seata-config-core/src/main/java/io/seata/config/FileConfigFactory.java:49]: unnecessary

File: config/seata-config-core/src/main/java/io/seata/config/FileConfigFactory.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import java.io.File;
+import java.util.LinkedHashMap;
+import java.util.Set;
+
+/**
+ * @author wangwei-ying
+ */
+public class FileConfigFactory {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FileConfigFactory.class);
```
Comment (a364176773) [inline config/seata-config-core/src/main/java/io/seata/config/FileConfigFactory.java:32]: not in use

File: config/seata-config-core/src/main/java/io/seata/config/FileConfigFactory.java (hunk 1 above)
Comment (wangwei-ying) [reply config/seata-config-core/src/main/java/io/seata/config/FileConfigFactory.java:49]: construct??

Comment (a364176773) [reply config/seata-config-core/src/main/java/io/seata/config/FileConfigFactory.java:49]: > construct??

newline

File: config/seata-config-core/src/main/java/io/seata/config/FileConfigFactory.java (hunk 2 above)
Comment (wangwei-ying) [reply config/seata-config-core/src/main/java/io/seata/config/FileConfigFactory.java]: done

**Comment by a364176773** [comment]
I think you should add an example here
seata/script/server

**Comment by wangwei-ying** [comment]
> I think you should add an example here
> seata/script/server

done

File: script/server/config/file.properties [hunk 3]
```diff
@@ -0,0 +1,27 @@
+store.model=file
```
Comment (a364176773) [inline script/server/config/file.properties:1]: store.mode

File: script/server/config/file.properties [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+store.db.minConn=5
+store.db.maxConn=100
+store.db.globalTable=global_table
+store.db.branchTable=branch_table
+store.db.lockTable=lock_table
+store.db.queryLimit=100
+store.db.maxWait=5000
+store.redis.host=127.0.0.1
+store.redis.port=6379
+store.redis.password=
```
Comment (a364176773) [inline script/server/config/file.properties:23]: = ""

File: script/server/config/file.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
+
+## transaction log store, only used in seata-server
+store:
+
+  ## store mode: file、db、redis
+  model: file
```
Comment (a364176773) [inline script/server/config/file.yml:6]: mode

File: script/server/config/file.yml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    globalTable: global_table
+    branchTable: branch_table
+    lockTable: lock_table
+    queryLimit: 100
+    maxWait: 5000
+  ## redis store property
+  redis:
+    host: 127.0.0.1
+    port: 6379
+    password:
```
Comment (a364176773) [inline script/server/config/file.yml:42]: : ""

File: script/server/config/registry.properties [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+config.apollo.apolloMeta=http://192.168.1.204:8801
+config.apollo.namespace=application
+config.apollo.apolloAccesskeySecret=
+config.zk.serverAddr=127.0.0.1:2181
+config.zk.sessionTimeout=6000
+config.zk.connectTimeout=2000
+config.zk.username=
+config.zk.password=
+config.etcd3.serverAddr=http://localhost:2379
+config.file.name=file.conf
```
Comment (a364176773) [inline script/server/config/registry.properties:54]: file.properties

File: script/server/config/registry.properties [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+registry.etcd3.serverAddr=http://localhost:2379
+registry.sofa.serverAddr=127.0.0.1:9603
+registry.sofa.application=default
+registry.sofa.region=DEFAULT_ZONE
+registry.sofa.datacenter=DefaultDataCenter
+registry.sofa.cluster=default
+registry.sofa.group=SEATA_GROUP
+registry.sofa.addressWaitTime=3000
+registry.file.name=file.conf
+config.type=file
```
Comment (a364176773) [inline script/server/config/registry.properties:37]: file.properties

File: script/server/config/registry.yml [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+    serverAddr: 127.0.0.1:9603
+    application: default
+    region: DEFAULT_ZONE
+    datacenter: DefaultDataCenter
+    cluster: default
+    group: SEATA_GROUP
+    addressWaitTime: 3000
+  
+  file:
+    name: file.conf
```
Comment (a364176773) [inline script/server/config/registry.yml:54]: file.yml

File: script/server/config/registry.yml [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+    sessionTimeout: 6000
+    connectTimeout: 2000
+    username: 
+    password: 
+  
+  etcd3:
+    serverAddr: http://localhost:2379
+  
+  file:
+    name: file.conf
```
Comment (a364176773) [inline script/server/config/registry.yml:89]: file.yml

File: config/seata-config-core/src/main/java/io/seata/config/file/YamlFileConfig.java [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+            Map config = configMap;
+            String[] dataId = path.split("\\.");
+            for (int i = 0; i < dataId.length - 1; i++) {
+                if (config.containsKey(dataId[i])) {
+                    config = (Map) config.get(dataId[i]);
+                } else {
+                    return null;
+                }
+            }
+            return String.valueOf(config.get(dataId[dataId.length - 1]));
```
Comment (a364176773) [inline config/seata-config-core/src/main/java/io/seata/config/file/YamlFileConfig.java:60]: config.get(dataId[dataId.length - 1]) If it is null  String.valueof(null)  to be  "null"

File: script/server/config/file.properties (hunk 4 above)
Comment (wangwei-ying) [reply script/server/config/file.properties:23]: `store.redis.password= ""`  This config value is  """"

**Review by a364176773 — APPROVED** [review]
LGTM

File: config/seata-config-core/src/test/resources/registry-testyaml.yml [hunk 12]
```diff
@@ -0,0 +1,11 @@
+registry:
```
Comment (l81893521) [inline config/seata-config-core/src/test/resources/registry-testyaml.yml:1]: Why not change the name to
```
registr-test-yaml.yml
```

File: script/server/config/registry.yml [hunk 13]
```diff
@@ -0,0 +1,91 @@
+registry:
+  # file 、nacos 、eureka、redis、zk、consul、etcd3、sofa
```
Comment (l81893521) [inline script/server/config/registry.yml:2]: Please check 
```
、
```

**Review by l81893521 — COMMENTED** [review]
Test fail. Can not load yml file that with space in the path

File: config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java [hunk 14]
```diff
@@ -139,8 +121,68 @@ public FileConfiguration(String name, boolean allowDynamicRefresh) {
...
+        String filePath = null;
+        boolean filePathCustom = name.startsWith(SYS_FILE_RESOURCE_PREFIX);
+        if (filePathCustom) {
+            filePath = name.substring(SYS_FILE_RESOURCE_PREFIX.length());
+        } else {
+            // projectDir first
+            filePath = this.getClass().getClassLoader().getResource("").getPath() + name;
+        }
+
+        File targetFile = new File(filePath);
```
Comment (l81893521) [inline config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java:141]: still test fail when file path include sapce mark.
```
/D:/Program%20Files/github/seata/server/target/classes/registry
```

**Review by l81893521 — APPROVED** [review]
LGTM

**Pull request merged**